### PR TITLE
Add possibility to duplicate object group

### PR DESCRIPTION
**Pull request by AlexandreSi**
image

image

Also:
- Use React sortable virtualized list to render object groups list
- Match group context menu with object one

### CODE REVIEW

**Review by ClementPasteau — COMMENTED** [review]
lgtm 🎉 
I'll let someone else approve, just because I'm not familiar with this part

File: newIDE/app/src/ObjectGroupsList/index.js [hunk 1]
```diff
@@ -231,18 +154,51 @@ export default class GroupsListContainer extends React.Component<Props, State> {
...
+    const newGroup = container.insertNew(
+      newName,
+      container.getPosition(group.getName()) + 1
+    );
+
+    unserializeFromJSObject(
+      newGroup,
+      serializeToJSObject(group),
+      'unserializeFrom'
     );
```
Comment (D8H) [inline newIDE/app/src/ObjectGroupsList/index.js:188]: It's probably the way it's done in other places in the code, but I find it strange to rely on serialization to copy the attributes of the group. Shouldn't containers have a `duplicate` method instead?

File: newIDE/app/src/ObjectGroupsList/index.js [hunk 2]
```diff
@@ -340,45 +332,59 @@ export default class GroupsListContainer extends React.Component<Props, State> {
...
-      key: 'add-groups-row',
-      object: null,
-    });
 
     // Force List component to be mounted again if globalObjectGroups or objectGroups
     // has been changed. Avoid accessing to invalid objects that could
     // crash the app.
     const listKey = objectGroups.ptr + ';' + globalObjectGroups.ptr;
 
+    const renamedGroupWithContext = fullList.find(
```
Comment (4ian) [inline newIDE/app/src/ObjectGroupsList/index.js:345]: Maybe useless to run the find if the group with context is null?

**Review by 4ian — APPROVED** [review]

File: newIDE/app/src/ObjectGroupsList/index.js (hunk 1 above)
Comment (AlexandreSi) [reply newIDE/app/src/ObjectGroupsList/index.js]: I guess so but it kinda grows the scope of this PR 😅 
It's an improvement we could do later maybe? Wait for another time we need a duplicate action?

**Comment by AlexandreSi** [comment]
Added in the last commits:
- Allow drop of local items on first displayed global item to put local items at the end of the local list (or right after last displayed local item if list is filtered)

**Pull request merged**